hazelcast-client/src/main/java/com/hazelcast/client/map/impl/nearcache/ClientHeapNearCache.java

```diff
@@ -260,4 +196,71 @@ public NearCacheStatsImpl getNearCacheStats() {
         stats.setOwnedEntryMemoryCost(ownedEntryMemoryCost);
         return stats;
     }
+
+    private void fireCacheEviction() {
+        if (canEvict.compareAndSet(true, false)) {
+            try {
+                executionService.executeInternal(new Runnable() {
+                    public void run() {
+                        try {
+                            Set<NearCacheRecord> records = new TreeSet<NearCacheRecord>(selectedComparator);
+                            records.addAll(cache.values());
+                            int evictSize = (int) (cache.size() * EVICTION_FACTOR);
+                            int i = 0;
+                            for (NearCacheRecord record : records) {
+                                cache.remove(record.getKey());
+                                stats.incrementEvictions();
+                                if (++i > evictSize) {
```

Comment:
We might have an off-by-one error here, since we remove one entry more than the calculated `evictSize`. So when there are 100 entries in the Near Cache the `evictSize` will be `20`, but `21` entries will be removed from the Near Cache. I adapted the tests accordingly, but this was a bit confusing to figure out :)